### ISSUE TITLE
Web app is missing page landmarks needed for screen reader navigation

### ISSUE THREAD

**Issue by hs4man21** [bluesky-social/social-app#646]
**Describe the bug**
The web app is lacking [page landmarks](https://a11y-101.com/development/landmarks) necessary for [the screen reader rotor](https://build.washingtonpost.com/resources/accessibility/screen-readers#VoiceOver%20(MacOS%20and%20iOS)) (which many users rely on to jump around a page without having to arrow through page elements one-by-one).

In particular, there is a need for `nav` elements that screen reader users can jump between for faster navigation on the site. The feed itself should also have a page landmark (see below for more on Twitter's implementation).

**To Reproduce**

1. Run the web app locally.
2. Turn on the screen reader VoiceOver with `cmd` + `F5` on a Mac, and open the rotor by using `space` + `U` (may vary based on your configuration).
3. Now, you can use the arrow keys to toggle menus as shown below. Note the only page landmark is for the search field.

**Expected behavior**

We would expect to see a number of landmarks available when using the VoiceOver rotor. Use Twitter as an example. There is a landmark for each navigation as well as the feed itself. This makes it much easier for folks to hop around the page as needed.

https://github.com/bluesky-social/social-app/assets/38192823/41a16efa-6f66-4f97-9253-df20f9ec292f

**Screenshots**
https://github.com/bluesky-social/social-app/assets/38192823/687e9b69-d432-4f8e-b425-07230750518f

**Details**

- Platform: Desktop Chrome MacOS
- Platform version: MacOS: Ventura 13.1, Chrome: 112.0.5615.137
- App version: 1.28(?), not sure where to find web app version number
